File: src/main/java/com/gitlab/daring/image/command/CommandScriptUtils.java
package com.gitlab.daring.image.command;

import org.bytedeco.javacpp.opencv_core.Mat;

import static org.apache.commons.lang3.StringUtils.join;

public class CommandScriptUtils {

    public static ScriptCommand parseScript(String script) {
        return CommandRegistry.Instance.parseScript(script);
    }

    public static void runScript(CommandEnv env, String script) {
        parseScript(script).execute(env);
    }

    public static String commandString(String cmd, Object... args) {
        return cmd + "(" + join(args, ", ") + ");";
    }

    public static Mat runCommand(CommandEnv env, String cmd, Object... args) {
        runScript(env, commandString(cmd, args));
        return env.getMat();
    }

    private CommandScriptUtils() {
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/CommandScriptUtils.kt
package com.gitlab.daring.image.command

import com.gitlab.daring.image.MainContext.commandRegistry

object CommandScriptUtils {

    @JvmStatic
    fun parseScript(script: String): ScriptCommand {
        return commandRegistry.parseScript(script)
    }

    fun runScript(env: CommandEnv, script: String) {
        parseScript(script).execute(env)
    }

    fun commandString(cmd: String, vararg args: Any): String {
        return "$cmd(${args.joinToString("," )});"
    }

    fun runCommand(env: CommandEnv, cmd: String, vararg args: Any) {
        runScript(env, commandString(cmd, *args))
    }

}